test/utils.jl

```diff
@@ -0,0 +1,178 @@
+using Manifolds
+
+using LinearAlgebra
+using Distributions
+using DoubleFloats
+using ForwardDiff
+using Random
+using ReverseDiff
+using StaticArrays
+using SimpleTraits
+using Test
+
+"""
+    test_manifold(m::Manifold, pts::AbstractVector;
+        test_forward_diff = true,
+        retraction_methods = [],
+        inverse_retraction_methods = [])
+
+Tests general properties of manifold `m`, given at least three different points
+that lie on it (contained in `pts`).
+
+# Arguments
+- `test_forward_diff = true`: if true, automatic differentiation using ForwardDiff is
+tested.
+- `retraction_methods = []`: retraction methods that will be tested.
+"""
+function test_manifold(M::Manifold, pts::AbstractVector;
```

Comment:
It's great to have this in such generality, I think `manifold_dimension` and ìnjectivity_radius` are missing, but might also be done one by one per manifold. Similarly `is_manifold_point`(also testing the errors that might occur) is missing and `is_tangent_vector`.

Reply:
and great that we have codecov running 👍

Reply:
I'd like to have the whole interface for manifolds tested here and so I expand it slowly. `manifold_dimension` and `injectivity_radius` are there, `is_manifold_point` and `is_tangent_vector` are partially tested (only the positive case). I think the negative case (when an error is thrown) should be tested separately for each manifold.

Reply:
Yes, I also think, that the negative cases can only be tested individually by carefully constructing invalid cases (if they even exist) and check for the right error messages.